### PR TITLE
[docs] Update the README

### PR DESCRIPTION
**Pull request by aravindhp**
- Fix the description of CatalogSourceConfig
- Add links to OLM docs
- Deprecate docs/how-to-upload-artifact.md

### CODE REVIEW

File: docs/how-to-upload-artifact.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # How To Push an Operator Artifact To an App Registry
 
+**THIS DOC AND THE SCRIPTS MENTIONED HERE HAVE BEEN DEPRECATED BY IN FAVOR OF A [NEW METHOD](https://github.com/operator-framework/community-operators/blob/master/README.md)**
```
Comment (kevinrizza) [inline docs/how-to-upload-artifact.md]: why not just delete the file then?

Comment (aravindhp) [reply docs/how-to-upload-artifact.md]: @tkashem says there is useful information for developers in here that he wants to retain. If that is not the case I am fine with deleting it.

Comment (kevinrizza) [reply docs/how-to-upload-artifact.md]: Personally I think that, if there is stuff in this we want to retain it should be moved out of this repo and into courier's docs.

Comment (aravindhp) [reply docs/how-to-upload-artifact.md]: @kevinrizza done

**Pull request merged**